reports-backend/src/main/java/org/jboss/da/reports/impl/ReportsGeneratorImpl.java

```diff
@@ -355,7 +355,14 @@ private void updateArtifact(ArtifactReport ar, Set<WhiteArtifact> allArtifacts)
                 } else {
                     different = Collections.emptySet();
                 }
+
                 differentVersion.put(gav, different);
+                for (Iterator<ProductArtifact> iterator = different.iterator(); iterator.hasNext();) {
+                    ProductArtifact next = iterator.next();
+                    String type = VersionComparator.difference(gav.getVersion(),
+                            next.getArtifact().getVersion()).toString();
+                    next.setDifferenceType(type);
+                }
```

Comment:
The method seems too big, could you move the code to a new prived method and add some javadoc to it?